src/tensorlake/functions_sdk/graph.py

```diff
@@ -234,6 +235,14 @@ def definition(self) -> ComputeGraphMetadata:
                     delay_multiplier=start_node.retries.delay_multiplier,
                 )
             ),
+            cache_key=(
+                CacheKeyMetadata(
+                    algorithm=start_node.cache_key["algorithm"],
```

Comment:
Let's not use dicts/maps and use dataclasses or pydantic models. Python is dynamic enough already, let's not add more without a need.